black-mulberry-core/src/main/java/com/black/mulberry/core/mapper/UserRegistrationMapper.java

```diff
@@ -2,21 +2,21 @@
 
 import com.black.mulberry.core.entity.User;
 import com.black.mulberry.core.mapper.base.BaseMapper;
-import com.black.mulberry.data.transfer.request.UserSignUpRequest;
+import com.black.mulberry.data.transfer.request.UserRegistrationRequest;
 import com.black.mulberry.data.transfer.response.UserResponse;
 import lombok.RequiredArgsConstructor;
 import org.modelmapper.ModelMapper;
 import org.springframework.stereotype.Service;
 
 @Service
 @RequiredArgsConstructor
-public class UserSignUpMapper implements BaseMapper<User, UserSignUpRequest, UserResponse> {
+public class UserRegistrationMapper implements BaseMapper<User, UserRegistrationRequest, UserResponse> {
```

Comment:
i think registration reguest DATA will be different from simple user reguest DATA . Reconsider the creation of a simple user request and each object of use for its intended purpose.
you can use overloading too !!!